test/powershell/Modules/Microsoft.PowerShell.Management/Set-Content.Tests.ps1

```diff
@@ -1,51 +1,102 @@
 Describe "Set-Content cmdlet tests" -Tags "CI" {
-  $file1 = "file1.txt"
-  Setup -File "$file1" -Content $file1
-  Context "Set-Content should actually set content" {
-    It "should set-Content of testdrive:\$file1" {
-      $result=set-content -path testdrive:\$file1 -value "ExpectedContent" -passthru 
-      $result| Should be "ExpectedContent"
+    BeforeAll {
+        $file1 = "file1.txt"
+        $filePath1 = join-path $testdrive $file1
+        # if the registry doesn't exist, don't run those tests
+        $skipRegistry = ! (test-path hklm:/)
+        #if not Admin, don't run these tests
+        if($IsWindows)
+        {
+            $windowsIdentity = [Security.Principal.WindowsIdentity]::GetCurrent()
+            $windowsPrincipal = new-object 'Security.Principal.WindowsPrincipal' $windowsIdentity
+            $SkipNotAdmin = $windowsPrincipal.IsInRole("Administrators") -eq $false
```

Comment:
The test architecture is that tests which require admin will only be run if the tag is proper. This test has to be tagged properly or it won't _ever_ be run. Because of that, you need to put it in it's own `Describe` block, even if it does not make sense logically